addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -327,8 +328,8 @@ def _timesheet_preprocess(self, vals_list):
             if not vals.get('project_id'):
                 vals['project_id'] = data.project_id.id
             if not vals.get('account_id'):
-                account_id = data.analytic_account_id.id
-                account = account_per_id[account_id]
+                account_id = data.analytic_account_id.id or data.project_id.analytic_account_id.id
+                account = account_per_id.get(account_id)
```

Comment:
```suggestion
                account = data._get_task_analytic_account_id() if vals.get('task_id') else data.analytic_account_id
```
Will do the trick in stable :slightly_smiling_face: